src/app/frontend/common/components/ingressrulelist/template.html

```diff
@@ -0,0 +1,92 @@
+<!--
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<kd-card [initialized]="initialized"
+         role="table">
+  <div title
+       i18n>Rules</div>
+
+  <div description>
+    <div class="kd-inline-property"
+         *ngIf="ingressSpecRules?.length">
+      <span class="kd-muted-light"
+            i18n>Items:&nbsp;</span>
+      <span>{{ingressSpecRules.length}}</span>
+    </div>
+  </div>
+
+  <div content
+       [hidden]="ingressSpecRules?.length === 0">
+    <mat-table [dataSource]="getDataSource()"
+               class="kd-table-no-footer">
+      <ng-container [matColumnDef]="getIngressRulesFlatColumns()[0]">
+        <mat-header-cell *matHeaderCellDef
+                         i18n>Host</mat-header-cell>
+        <mat-cell *matCellDef="let ingressRuleFlat">
+          <ng-container *ngIf="ingressRuleFlat.host">{{ingressRuleFlat.host}}</ng-container>
+          <ng-container *ngIf="!ingressRuleFlat.host">*</ng-container>
+        </mat-cell>
+      </ng-container>
+      <ng-container [matColumnDef]="getIngressRulesFlatColumns()[1]">
+        <mat-header-cell *matHeaderCellDef
+                         i18n>Path</mat-header-cell>
+        <mat-cell *matCellDef="let ingressRuleFlat">
+          <ng-container *ngIf="ingressRuleFlat.path.path">{{ingressRuleFlat.path.path}}</ng-container>
```

Comment:
When it is uncertain if the property will exist use `ingressRuleFlat?.path?.path` to avoid undefined exceptions.

Second thing is that for properties that are often empty try to provide some default value like
```html
{{!!ingressRuleFlat?.path?.path ? ingressRuleFlat?.path?.path : '-'}}
```